lib/db/dao/related_topic_dao.dart

```diff
@@ -20,4 +21,24 @@ class RelatedTopicDao {
         .map((el) => new RelatedTopic.fromMap(el))
         .toList(growable: false);
   }
+
+  Future<List<Topic>> getTopicsByRelatedTopicId(String topicId,
+      {Database db}) async {
+    db = db ?? await new AppDatabase().getDb();
+    List<Map> maps =
+        await db.query('${Topic.table} t, ${RelatedTopic.table} r', columns: [
+      't.${Topic.idCol}',
+      't.${Topic.nameCol}',
+      't.${Topic.imageCol}',
+      //  't.${Topic.colorCol}'
```

Comment:
The entity object should always have the full data. Whatever is in the database, populate it on the entity object